### PR TITLE
Fix some runtime exception 

### PR DESCRIPTION
**Pull request by steelgeek091**

### CODE REVIEW

File: crates/rooch-types/src/framework/session_key.rs [hunk 1]
```diff
@@ -30,12 +30,24 @@ pub struct SessionScope {
...
 
 impl SessionScope {
-    pub fn new(module_address: AccountAddress, module_name: &str, function_name: &str) -> Self {
-        Self {
+    pub fn new(
+        module_address: AccountAddress,
+        module_name: &str,
+        function_name: &str,
+    ) -> Result<Self> {
+        let module_name_value = match MoveAsciiString::from_str(module_name) {
```
Comment (jolestar) [inline crates/rooch-types/src/framework/session_key.rs:38]: use `?` or `map_err` here?

File: frameworks/moveos-stdlib/src/natives/moveos_stdlib/account.rs [hunk 2]
```diff
@@ -39,8 +40,13 @@ pub fn native_create_signer(
...
 ) -> PartialVMResult<NativeResult> {
-    debug_assert!(ty_args.is_empty());
-    debug_assert!(arguments.len() == 1);
+    if !ty_args.is_empty() {
+        return args_count_error(gas_params.base);
+    }
+
+    if arguments.len() != 1 {
+        return args_count_error(gas_params.base);
+    }
```
Comment (jolestar) [inline frameworks/moveos-stdlib/src/natives/moveos_stdlib/account.rs:49]: Does the `debug_assert!` need to be changed? I think if the Move native function definition is corrected, the verifier should ensure the caller can not pass the wrong args.

Comment (steelgeek091) [reply frameworks/moveos-stdlib/src/natives/moveos_stdlib/account.rs:49]: I attempted to bypass argument and type argument validation. While I could bypass it during compilation, but the verifier returns errors during module deployment.
I will remove the unnecessary validation code.

File: frameworks/moveos-stdlib/src/natives/moveos_stdlib/hash.rs [hunk 3]
```diff
@@ -21,8 +22,13 @@ fn hash<H: HashFunction<DIGEST_SIZE>, const DIGEST_SIZE: usize>(
...
 ) -> PartialVMResult<NativeResult> {
-    debug_assert!(ty_args.is_empty());
-    debug_assert!(args.len() == 1);
+    if !ty_args.is_empty() {
+        return args_count_error(gas_params.base);
+    }
+
+    if args.len() != 2 {
+        return args_count_error(gas_params.base);
+    }
```
Comment (jolestar) [inline frameworks/moveos-stdlib/src/natives/moveos_stdlib/hash.rs:31]:
```
args.len() != 1
```
This code break the test

```
2024-04-15T04:49:31.044327Z ERROR rooch_relayer::actor::relayer: Relayer execute relay block(hash: 743d2520162f43eeb42052f2158d2c73a0f16e1f4b274e21c186ab167990fe68, height: 2) failed, status: ABORTED { code: 13, location: 0000000000000000000000000000000000000000000000000000000000000002::hash }
```

Comment (steelgeek091) [reply frameworks/moveos-stdlib/src/natives/moveos_stdlib/hash.rs]: All unnecessary code for argument and type argument validation has been removed.

**Review by jolestar — APPROVED** [review]

**Pull request merged**